### PR TITLE
[FIX] vscode: fix addons detection irrelevant files in odoo path

### PR DESCRIPTION
**Pull request by livaios**

### CODE REVIEW

File: package-lock.json
Comment (fda-odoo) [inline package-lock.json:1]: Ha, I think we can add package_lock in the gitignore file :)

File: vscode/client/extension.ts [hunk 1]
```diff
@@ -327,36 +327,39 @@ function deleteOldFiles(context: ExtensionContext, outputChannel: OutputChannel)
...
+                    break
+            }
         }
     }
 }
 
 async function checkOdooPath(context: ExtensionContext) {
     let currentConfig = getCurrentConfig(context)
-    let odooFound = workspace.getWorkspaceFolder(Uri.parse(currentConfig.odooPath))
+    let odooFound = currentConfig ? workspace.getWorkspaceFolder(Uri.parse(currentConfig.odooPath)) : true
```
Comment (fda-odoo) [inline vscode/client/extension.ts:362]: Why true if do not find the currentConfig?

File: vscode/client/extension.ts [hunk 2]
```diff
@@ -327,36 +327,39 @@ function deleteOldFiles(context: ExtensionContext, outputChannel: OutputChannel)
...
-            case ("Update current configuration"):
-                ConfigurationWebView.render(context, currentConfig.id);
-                break
-            case ("View Paths"):
-                odooOutputChannel.show();
-                break
+    if (currentConfig) {
+        let missingFiles = files.filter(file => {
+            return !currentConfig.addons.some((addon) =>
+                file.fsPath.startsWith(addon) || file.fsPath.startsWith(currentConfig.odooPath)
```
Comment (fda-odoo) [inline vscode/client/extension.ts:333]: Why `file.fsPath.startsWith(currentConfig.odooPath)` is in the `some` but is not using addon?
`return !currentConfig.addons.some((addon) =>file.fsPath.startsWith(addon)) || !file.fsPath.startsWith(currentConfig.odooPath` is maybe a little more readable

**Review by fda-odoo — CHANGES_REQUESTED** [review]

File: package-lock.json
Comment (livaios) [reply package-lock.json:1]: Ahh my bad I thought it was already there

File: vscode/client/extension.ts (hunk 1 above)
Comment (livaios) [reply vscode/client/extension.ts:362]: not finding the currentConfig means that the extension is disabled so we will not check for the odoo path to match

**Pull request merged**